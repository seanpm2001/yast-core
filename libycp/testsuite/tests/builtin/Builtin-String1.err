Parsed:
----------------------------------------------------------------------
mergestring ([], ".")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
mergestring (["abc", "dev", "ghi"], "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
mergestring (["abc", "dev", "ghi"], "123")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** deletechars **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
deletechars ("1a2b3c", "abc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
deletechars ("1a2b3c", "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
deletechars ("", "abc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
deletechars ("", "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** filterchars **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
filterchars ("1a2b3c", "abc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
filterchars ("1a2b3c", "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
filterchars ("", "abc")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
filterchars ("", "")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** crypt  -  disabled **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** tolower **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tolower ("aBcDeF")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** find et.al. **"
----------------------------------------------------------------------
[Parser] tests/builtin/Builtin-String1.ycp:43 Warning: find(...) is deprecated, please fix
Parsed:
----------------------------------------------------------------------
find ("abcdefghi", "efg")
----------------------------------------------------------------------
[Parser] tests/builtin/Builtin-String1.ycp:44 Warning: find(...) is deprecated, please fix
Parsed:
----------------------------------------------------------------------
find ("aaaaa", "z")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
search ("abcdefghi", "efg")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
search ("aaaaa", "z")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findfirstnotof ("abcdefghi", "abcefghi")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findfirstnotof ("aaaaa", "a")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findfirstof ("abcdefghi", "cxdv")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findfirstof ("aaaaa", "z")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findlastof ("abcdecfghi", "cxdv")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
findlastof ("aaaaa", "z")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** regex **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpmatch ("aaabbb", "ab") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpmatch ("aaabbb", "ba") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpmatch ("aaabbb", ".*ab.*") == true)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpmatch ("aaabbb", ".*ba.*") == false)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(.*ab).*", "s_\\1_e") == "s_aaab_e")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(.*abb)(.*)", "s_\\1_e\\2") == "s_aaabb_eb")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(a*b)(.*)", "\\1") == "aaab")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(a*b)(.*)", "\\1\\1") == "aaabaaab")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(a*b)(.*)", "\\") == "\\")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(a*b)(.*)", "\\\\") == "\\\\")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(a*b)(.*)", "A\\NZ") == "A\\NZ")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("aaabbb", "(a*b)(.*)", "A\\\\1Z") == "A\\aaabZ")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexpsub ("machine=i386\\2ci486\\2ci586\\2ci686", "^([^=]*)=(.*)", "\\2") == "i386\\2ci486\\2ci586\\2ci686")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexptokenize ("aaabbBb", "(.*[[:upper:]]).*") == ["aaabbB"])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexptokenize ("aaabbb", "(.*ab)(.*)") == ["aaab", "bb"])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexptokenize ("aaabbb", "(.*ba).*") == [])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
(regexptokenize ("aaabbb", "(.*ba).*()") == [])
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** sformat **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
sformat ("%2 is greater %% than %1", 3, "five")
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
"** conversions **"
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tostring (tofloat (63), 4)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tostring (0.12345, 4)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tohexstring (31)
----------------------------------------------------------------------
Parsed:
----------------------------------------------------------------------
tohexstring (31, 4)
----------------------------------------------------------------------
