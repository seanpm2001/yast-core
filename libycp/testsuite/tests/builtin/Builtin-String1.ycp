
# ---------------------------------------------------------
#
#  Filename:	Builtin-String.ycp
#
#  Purpose:	test cases for all string builtins
#
# ---------------------------------------------------------

(mergestring ([], "."))
(mergestring (["abc", "dev", "ghi"], ""))
(mergestring (["abc", "dev", "ghi"], "123"))


("** deletechars **")

(deletechars ("1a2b3c", "abc"))
(deletechars ("1a2b3c", ""))
(deletechars ("", "abc"))
(deletechars ("", ""))


("** filterchars **")

(filterchars ("1a2b3c", "abc"))
(filterchars ("1a2b3c", ""))
(filterchars ("", "abc"))
(filterchars ("", ""))


("** crypt  -  disabled **")

//(crypt ("readable"))
//(cryptmd5 ("readable"))


("** tolower **")

(tolower ("aBcDeF"))

("** find et.al. **")

(find ("abcdefghi", "efg"))
(find ("aaaaa", "z"))

(search ("abcdefghi", "efg"))
(search ("aaaaa", "z"))

(findfirstnotof ("abcdefghi", "abcefghi"))
(findfirstnotof ("aaaaa", "a"))
(findfirstof ("abcdefghi", "cxdv"))
(findfirstof ("aaaaa", "z"))
(findlastof ("abcdecfghi", "cxdv"))
(findlastof ("aaaaa", "z"))


("** regex **")

(regexpmatch ("aaabbb", "ab") == true)
(regexpmatch ("aaabbb", "ba") == false)
(regexpmatch ("aaabbb", ".*ab.*") == true)
(regexpmatch ("aaabbb", ".*ba.*") == false)

(regexpsub ("aaabbb", "(.*ab).*", "s_\\1_e") == "s_aaab_e")
(regexpsub ("aaabbb", "(.*abb)(.*)", "s_\\1_e\\2") == "s_aaabb_eb")
// bnc#552914
(regexpsub ("aaabbb", "(a*b)(.*)", "\\1") == "aaab")
(regexpsub ("aaabbb", "(a*b)(.*)", "\\1\\1") == "aaabaaab")
(regexpsub ("aaabbb", "(a*b)(.*)", "\\") == "\\")
(regexpsub ("aaabbb", "(a*b)(.*)", "\\\\") == "\\\\")
(regexpsub ("aaabbb", "(a*b)(.*)", "A\\NZ") == "A\\NZ")
(regexpsub ("aaabbb", "(a*b)(.*)", "A\\\\1Z") == "A\\aaabZ")
(regexpsub ("machine=i386\\2ci486\\2ci586\\2ci686", "^([^=]*)=(.*)", "\\2") == "i386\\2ci486\\2ci586\\2ci686")

(regexptokenize ("aaabbBb", "(.*[[:upper:]]).*") == [ "aaabbB" ])
(regexptokenize ("aaabbb", "(.*ab)(.*)") == ["aaab", "bb"])
(regexptokenize ("aaabbb", "(.*ba).*") == [])
(regexptokenize ("aaabbb", "(.*ba).*()") == [])


("** sformat **")

(sformat ("%2 is greater %% than %1", 3, "five"))


("** conversions **")

(tostring (tofloat (63), 4))
(tostring (0.12345, 4))
(tohexstring (31))
(tohexstring (31, 4))

