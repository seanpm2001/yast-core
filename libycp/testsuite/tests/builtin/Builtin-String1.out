("")
("abcdevghi")
("abc123dev123ghi")
("** deletechars **")
("123")
("1a2b3c")
("")
("")
("** filterchars **")
("abc")
("")
("")
("")
("** crypt  -  disabled **")
("** tolower **")
("abcdef")
("** find et.al. **")
(4)
(-1)
(4)
(nil)
(3)
(nil)
(2)
(nil)
(5)
(nil)
("** regex **")
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
(true)
("** sformat **")
("five is greater % than 3")
("** conversions **")
("63.0000")
("0.1235")
("0x1f")
("0x001f")
